sdk/communication/Azure.Communication.Administration/src/Models/ReleasePhoneNumberOperation.cs

```diff
@@ -0,0 +1,159 @@
+
+        /// <inheritdocs />
+        public override bool HasValue => (_rawResponse != null || _finalRawResponse != null) && _value != null;
+
+        /// <inheritdocs />
+        public override Response GetRawResponse() => HasCompleted ? _finalRawResponse : _rawResponse;
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        public override Response UpdateStatus(CancellationToken cancellationToken = default)
+            => UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+            => await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <param name="async" />
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
+        {
+            if (HasCompleted)
+            {
+                return GetRawResponse();
+            }
+
+            if (cancellationToken == default)
+            {
+                cancellationToken = _cancellationToken;
+            }
+
+            Response<PhoneNumberRelease> update = async
+                ? await _client.GetReleaseByIdAsync(releaseId: Id, cancellationToken: cancellationToken).ConfigureAwait(false)
+                : _client.GetReleaseById(releaseId: Id, cancellationToken: cancellationToken);
+
+            var terminateStatuses = new ReleaseStatus[]
+            {
+                ReleaseStatus.Complete,
+                ReleaseStatus.Failed,
+                ReleaseStatus.Expired
+            };
+
+            Response response = update.GetRawResponse();
+            _rawResponse = response;
+
+            if (update.Value.Status.HasValue && terminateStatuses.Contains(update.Value.Status.Value))
+            {
+                lock (_lockObject)
```

Comment:
Same comments as for the other LROs, I think we can only keep one `_response` field and probably don't need the lock,